test/core/names.wast

```diff
@@ -18,52 +18,152 @@
-(assert_return (invoke "0") (f32.const 0x1.97p+2))
-(assert_return (invoke "_") (f32.const 0x1.98p+2))
-(assert_return (invoke "$") (f32.const 0x1.99p+2))
-(assert_return (invoke "@") (f32.const 0x2.00p+2))
-(assert_return (invoke "NaN") (f32.const 0x2.01p+2))
-(assert_return (invoke "Infinity") (f32.const 0x2.02p+2))
-(assert_return (invoke "if") (f32.const 0x2.03p+2))
-(assert_return (invoke "﻿") (f32.const 0x2.04p+2))
+(assert_return (invoke "") (i32.const 0))
+(assert_return (invoke "0") (i32.const 1))
+(assert_return (invoke "_") (i32.const 2))
+(assert_return (invoke "$") (i32.const 3))
+(assert_return (invoke "@") (i32.const 4))
+(assert_return (invoke "~!@#$%^&*()_+`-={}|[]\\:\";'<>?,./ ") (i32.const 5))
+(assert_return (invoke "NaN") (i32.const 6))
+(assert_return (invoke "Infinity") (i32.const 7))
+(assert_return (invoke "if") (i32.const 8))
+(assert_return (invoke "malloc") (i32.const 9))
+(assert_return (invoke "_malloc") (i32.const 10))
+(assert_return (invoke "__malloc") (i32.const 11))
+(assert_return (invoke "a") (i32.const 12))
+(assert_return (invoke "A") (i32.const 13))
+(assert_return (invoke "﻿") (i32.const 14))
+(assert_return (invoke "Å") (i32.const 15))
+(assert_return (invoke "Å") (i32.const 16))
+(assert_return (invoke "Å") (i32.const 17))
+(assert_return (invoke "ﬃ") (i32.const 18))
+(assert_return (invoke "ffi") (i32.const 19))
+(assert_return (invoke "\00\01\02\03\04\05\06\07\08\09\0a\0b\0c\0d\0e\0f") (i32.const 20))
+(assert_return (invoke "\10\11\12\13\14\15\16\17\18\19\1a\1b\1c\1d\1e\1f") (i32.const 21))
+(assert_return (invoke " \7f") (i32.const 22))
+(assert_return (invoke "\c2\80\c2\81\c2\82\c2\83\c2\84\c2\85\c2\86\c2\87\c2\88\c2\89\c2\8a\c2\8b\c2\8c\c2\8d\c2\8e\c2\8f") (i32.const 23))
+(assert_return (invoke "\c2\90\c2\91\c2\92\c2\93\c2\94\c2\95\c2\96\c2\97\c2\98\c2\99\c2\9a\c2\9b\c2\9c\c2\9d\c2\9e\c2\9f") (i32.const 24))
+(assert_return (invoke "\ef\bf\b0\ef\bf\b1\ef\bf\b2\ef\bf\b3\ef\bf\b4\ef\bf\b5\ef\bf\b6\ef\bf\b7") (i32.const 25))
+(assert_return (invoke "\ef\bf\b8\ef\bf\b9\ef\bf\ba\ef\bf\bb\ef\bf\bc\ef\bf\bd\ef\bf\be\ef\bf\bf") (i32.const 26))
+(assert_return (invoke "‍") (i32.const 27))
+(assert_return (invoke "‌") (i32.const 28))
+(assert_return (invoke "￯​ ­⁠ ‮‭") (i32.const 29))
+(assert_return (invoke "‎‏‑  ‪‫‬ ⁦⁧⁨⁩") (i32.const 30))
+(assert_return (invoke "⁪⁫⁬⁭⁮⁯") (i32.const 31))
+(assert_return (invoke "⁡⁢⁣⁤") (i32.const 32))
+(assert_return (invoke "𐀀󟿿􏿿") (i32.const 33))
+(assert_return (invoke "Z̴͇̫̥̪͓͈͔͎̗̞̺̯̱̞̙̱̜̖̠̏͆̆͛͌͘͞ḁ̶̰̳̭͙̲̱̹̝͎̼͗ͨ̎̄̆͗̿̀́͟͡l̶̷͉̩̹̫̝͖̙̲̼͇͚͍̮͎̥̞̈́͊͗ͦ̈́ͫ̇́̚ͅͅg̶͕͔͚̩̓̐̅ͮ̔̐̎̂̏̾͊̍͋͊ͧ́̆ͦ͞o̡͋̔͐ͪͩ͏̢̧̫̙̤̮͖͙͓̺̜̩̼̘̠́") (i32.const 34))
+(assert_return (invoke "ᅟᅠㅤﾠ") (i32.const 35))
+(assert_return (invoke "︀") (i32.const 36))
+(assert_return (invoke "︄") (i32.const 37))
+(assert_return (invoke "󠄀") (i32.const 38))
+(assert_return (invoke "󠇯") (i32.const 39))
+(assert_return (invoke "̈") (i32.const 40))
```

Comment:
As a bonus, I've now added a few more tests, including some more interesting uses of the replacement character.